### PR TITLE
[cluster-test] Mint new accounts concurrently

### PR DESCRIPTION
**Pull request by ankushagarwal**
Depends on #1838 hence the two commits

## Summary

* We currently mint accounts sequentially for load test, because we only have 1 minting account
* In this change: We first create a series of seed accounts which we mint sequentially. After that, we mint new accounts from each of the seed accounts concurrently.

Also
* Refactored some functions by adding `is_faucet: bool` field to AccountData
* Added retry logic in execute_and_wait_transactions

## Testing 

With the new change, minting 10k accounts takes about 20 seconds. Without this PR, it takes about 50 seconds.

```
[ec2-user:validator@ip-10-0-2-127 ~]$ ct --emit-tx --accounts-per-client 100
Nov 27 19:10:12.838 INFO Discovered 100 peers
Nov 27 19:10:13.427 INFO Completed minting seed accounts
Nov 27 19:10:32.205 INFO Mint is done
Nov 27 19:11:13.261 INFO Tps: 471, latency: 10410 ms
Nov 27 19:11:23.275 INFO Tps: 456, latency: 10582 ms
```

Closes https://github.com/libra/libra/issues/1197

### CODE REVIEW

File: testsuite/cluster-test/src/tx_emitter.rs [hunk 1]
```diff
@@ -85,6 +87,26 @@ pub struct EmitJobRequest {
     pub thread_params: EmitThreadParams,
 }
 
+/// Create `num_new_accounts` by transferring libra from `source_account`. Return Vec of created
+/// accounts
```
Comment (andll) [inline testsuite/cluster-test/src/tx_emitter.rs]: Can you move this fn down to where other utility functions are?

**Comment by ankushagarwal** [comment]
I found embedding `is_faucet` inside `AccountData` helpful because it allows us to create `fn create_new_accounts` without passing around a boolean around in function parameters. I like the pattern that we have here because we are able to encapsulate both the `faucet -> N seed accounts` and `seed account -> N accounts` into a single function `create_new_accounts`.

I'm not convinced that `faucet -> root seed account -> N seed accounts -> actual accounts` (I am assuming that is what you meant) is necessarily simpler than `faucet -> N seed accounts -> actual accounts`

**Comment by andll** [comment]
You don't need 'root seed account', you can just mint money in faucet account itself?

**Comment by ankushagarwal** [comment]
Holding off merging till I get a chance to test this

**Pull request closed without merging**